dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -95,6 +95,23 @@ def self.download_pardot_prospects(last_id = nil, limit = nil)
     end
   end
 
+  def self.download_deleted_pardot_prospects(last_id = nil, limit = nil)
+    PardotV2.retrieve_prospects(last_id || 0,  %w(id email), limit, true) do |deleted_prospects|
+      current_time = Time.now.utc
+      batch = deleted_prospects.map do |item|
+        {
+          email: item['email'],
+          data_rejected_at: current_time,
+          data_rejected_reason: PardotHelpers::ERROR_PROSPECT_DELETED_FROM_PARDOT
+        }
+      end
+
+      import! batch,
+        validate: false,
+        on_duplicate_key_update: [:data_rejected_at, :data_rejected_reason]
```

Comment:
Why only these two fields?

Reply:
Correct, the bare minimum we want to save here is why an email shouldn't be synced to Pardot. No need to overwrite other fields such as `pardot_id` and `data_synced`.